### ISSUE TITLE
Feature request: Priority assignment correcting

### ISSUE THREAD

**Issue by soxrok2212** [s3inlc/hashtopussy#201]
When pre-conf tasks are set with a priority and there are already tasks with an arbitrary priority in the queue, creating a new task from the pre-confs with a set priority puts this one on top instead of finishing the current one up to finish. This way of assigning priority seems a bit backwards for me, unless there is some other kind of priority assignment system I'm not aware of (sorry for using "priority 1000 times).

**Comment by winxp5421** [comment]
Yes, i have thought about this many times and honestly, i cannot say which way is "correct" as i feel they are personal preference.

Doing current highest priority + pre-conf priority is just a sure way to get an unused priority value and its simple enough to add more and more as needed.

We could, for example, switch our definition of Highest priority to something like lowest numerical is ran first meaning a priority of 1 would be ran first and a priority of 2 would be ran second and so on. So the current way of handling pre-conf tasks would not have to be modified but, this way of thinking kind of goes against the more logical way of highest value first. ( the current setup )

The other option is to switch the way pre-conf tasks' priority is calculated something like  current running task has a priority of 100 you issue a pre-conf task with a priority of 10, 5, and 2 with the resulting tasks having a priority of 90, 85, 82 , however, there is a finite amount of room until you reach zero priority and then we would have to handle negative priority values etc. 

Ultimately this decision is s3in!c's  to make so i will refer to his judgment on this. Though I do see the value of adding pre-conf tasks under the current running tasks it is certainly harder to implement

**Comment by soxrok2212** [comment]
Or perhaps there could be a whole other option such as "add to bottom of queue"? Then you could still assign a higher priority if needed, but also the current running task would just be pushed up by 1 and the new task could be added in the bottom.
